exporter/elasticsearchexporter/bulkindexer.go

```diff
@@ -383,6 +484,81 @@ func flushBulkIndexer(
 		}
 		failedDocsInputLogger.Debug("failed to index document; input may contain sensitive data", fields...)
 	}
+	if stat.Indexed > 0 {
+		tb.ElasticsearchDocsIndexed.Add(
+			ctx,
+			stat.Indexed,
+			metric.WithAttributeSet(attribute.NewSet(
+				attribute.String("outcome", "success"),
+			)),
+		)
+	}
+	if tooManyReqs > 0 {
+		tb.ElasticsearchDocsIndexed.Add(
+			ctx,
+			tooManyReqs,
+			metric.WithAttributeSet(attribute.NewSet(
+				attribute.String("outcome", "too_many"),
+			)),
+		)
+	}
+	if clientFailed > 0 {
+		tb.ElasticsearchDocsIndexed.Add(
+			ctx,
+			clientFailed,
+			metric.WithAttributeSet(attribute.NewSet(
+				attribute.String("outcome", "failed_client"),
+			)),
+		)
+	}
+	if serverFailed > 0 {
+		tb.ElasticsearchDocsIndexed.Add(
+			ctx,
+			clientFailed,
+			metric.WithAttributeSet(attribute.NewSet(
+				attribute.String("outcome", "failed_server"),
+			)),
+		)
+	}
+	if stat.FailureStoreDocs.Used > 0 {
+		tb.ElasticsearchDocsIndexed.Add(
+			ctx,
+			stat.FailureStoreDocs.Used,
+			metric.WithAttributeSet(attribute.NewSet(
+				attribute.String("outcome", "failure_store"),
+				attribute.String("failure_store", string(docappender.FailureStoreStatusUsed)),
+			)),
+		)
+	}
+	if stat.FailureStoreDocs.Failed > 0 {
+		tb.ElasticsearchDocsIndexed.Add(
+			ctx,
+			stat.FailureStoreDocs.Failed,
+			metric.WithAttributeSet(attribute.NewSet(
+				attribute.String("outcome", "failure_store"),
+				attribute.String("failure_store", string(docappender.FailureStoreStatusFailed)),
+			)),
+		)
+	}
+	if stat.FailureStoreDocs.NotEnabled > 0 {
+		tb.ElasticsearchDocsIndexed.Add(
+			ctx,
+			stat.FailureStoreDocs.NotEnabled,
+			metric.WithAttributeSet(attribute.NewSet(
+				attribute.String("outcome", "failure_store"),
+				attribute.String("failure_store", string(docappender.FailureStoreStatusNotEnabled)),
+			)),
+		)
+	}
+	if stat.RetriedDocs > 0 {
+		tb.ElasticsearchDocsRetried.Add(
+			ctx,
+			stat.RetriedDocs,
+			metric.WithAttributeSet(attribute.NewSet(
+				attribute.Int("greatest_retry", stat.GreatestRetry),
```

Comment:
I'm not sure how useful this is, but this will likely be a high cardinality one. wdyt about removing it?

Reply:
Done in https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/41254/commits/c40a7d06797aac09cb57a4f3384b4b7ead4ff204